openmp/runtime/test/tasking/bug_nested_proxy_task.c

```diff
@@ -50,12 +50,21 @@ typedef struct kmp_depend_info {
      union {
         kmp_uint8 flag; // flag as an unsigned char
         struct { // flag as a set of 8 bits
-            unsigned in : 1;
-            unsigned out : 1;
-            unsigned mtx : 1;
-            unsigned set : 1;
-            unsigned unused : 3;
-            unsigned all : 1;
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
I was wondering whether this test something that is compiler(clang) specific, or is this portable?
Since kmp.h has the same code, it make sense to replicate the pattern here.

Reply:
> I was wondering whether this test something that is compiler(clang) specific, or is this portable? 

Checked compilers in [Compiler Explorer](https://godbolt.org/), the test works with gcc, clang, icc, icx, and zig cc. Does not seem to work with msvc.

Reply:
This is something that would also need to be addressed in kmp.h. 

Does this versioning have a correctness implication, or is this a performance optimization?
In the latter case, from my perspective, a viable approach would be hardcode the endianess, if the two macros are undefined.

It's your choice to fix it as part of this patch or whether you prefer to fix it in a separate pr:)